src/topp/GNPSExport.cpp

```diff
@@ -79,297 +147,368 @@ using namespace std;
+    output_file << "PEPMASS=" << feature_mz << "\n";
+    output_file << "FILE_INDEX=" << spec_index << "\n";
+    output_file << "RTINSECONDS=" << feature_rt << "\n";
+
+    for (const pair<double,int>& ms2 : ms2_block)
+    {
+      if (ms2.second > 0)
+      {
+        output_file << ms2.first << "\t" << ms2.second << "\n";
+      }
+    }
+
+    output_file << "END IONS" << "\n\n";
+  }
+
+  void generateMSMSSpectrumBins(
+    const vector<pair<double,int>>& sorted_mz_int_pairs,
+    double delta_mz,
+    map<double,int>& ms2_block
+  )
+  {
+    // generate new spectrum
+    vector<double> mz_merged;
+    vector<double> intensity_merged;
+
+    // double last_mz = numeric_limits<double>::min();
+    double last_mz = sorted_mz_int_pairs[0].first;    
+    double sum_mz = 0;
+    int sum_intensity = 0;
+    double count = 0;
+    for (const auto mz_int : sorted_mz_int_pairs)
+    {
+      if (abs(mz_int.first - last_mz) > delta_mz && count > 0)
+      {
+        if (sum_intensity > 0)
+        {
+          double mz_merged = sum_mz/count;
+          int int_merged = sum_intensity;
+          ms2_block[mz_merged] = int_merged;
+        }
+
+        last_mz = mz_int.first;
+        sum_mz = 0;
+        count = 0;
+        sum_intensity = 0;
+      }
+
+      sum_mz += mz_int.first;
+      sum_intensity += mz_int.second;
+      count++;
+    }
+    //remaining scans in last bucket
+    if (count > 0 && sum_intensity > 0)
+    {
+      double mz_merged = sum_mz/count;
+      double int_merged = sum_intensity;
+
+      ms2_block[mz_merged] = int_merged;
+    }
+
+    // return would be the reformatted map<double,int> ms2_block passed in by value
+  }
+
+  void sortElementMapsByIntensity(const ConsensusFeature& feature, vector<pair<int,double>>& element_maps)
+  {    
+    // convert element maps to vector of pair<int,double>(map, intensity)     
+    for (ConsensusFeature::HandleSetType::const_iterator feature_iter = feature.begin();\
+          feature_iter != feature.end(); ++feature_iter)
+    {
+      element_maps.emplace_back(feature_iter->getMapIndex(), feature_iter->getIntensity());
+    }
+
+    // sort elements by intensity
+    sort(element_maps.begin(), element_maps.end(), [](const pair<int,double> &a, const pair<int,double> &b)
+    {
+      return a.second > b.second;
+    });
+
+    // return value will be reformatted vector<int> element_maps passed in by value
+  }
+
+  void getElementPeptideIdentificationsByElementIntensity(
+    const ConsensusFeature& feature,
+    const vector<pair<int,double>>& sorted_element_maps,
+    vector<pair<int,int>>& pept_indices
+  )
+  {
+    for (const pair<int,double> element_pair : sorted_element_maps)
+    {
+      int element_map = element_pair.first;
+
+      vector<PeptideIdentification> feature_pepts = feature.getPeptideIdentifications();
+      for (const PeptideIdentification pept_id : feature_pepts)
```

Comment:
this is another loop copy error in the checker (I guess)
use `const PeptideIdentification& pept_id`